lib/phoenix_live_view/async.ex

```diff
@@ -50,20 +53,36 @@ defmodule Phoenix.LiveView.Async do
     :ok
   end
 
+  defp validate_function_env(func, op) do
+    {:env, variables} = Function.info(func, :env)
+
+    if Enum.any?(variables, &match?(%Phoenix.LiveView.Socket{}, &1)) do
```

Comment:
Should we also try to match on `%{__changed__: _}`?

Reply:
That would guard against something like

```elixir
assigns = socket.assigns

assign_async(socket, :foo, fn ->
  do_something(assigns)
end
```

I suppose?